src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinActivityConversionExtensions.cs

```diff
@@ -0,0 +1,219 @@
+
+            ZipkinEndpoint remoteEndpoint = null;
+            if ((activity.Kind == ActivityKind.Client || activity.Kind == ActivityKind.Producer) && attributeEnumerationState.RemoteEndpointServiceName != null)
+            {
+                remoteEndpoint = RemoteEndpointCache.GetOrAdd(attributeEnumerationState.RemoteEndpointServiceName, ZipkinEndpoint.Create);
+            }
+
+            var annotations = PooledList<ZipkinAnnotation>.Create();
+            ListEnumerator<ActivityEvent, PooledList<ZipkinAnnotation>>.AllocationFreeForEach(activity.Events, ref annotations, ProcessActivityEventsRef);
+
+            return new ZipkinSpan(
+                EncodeTraceId(context.TraceId, useShortTraceIds),
+                parentId,
+                EncodeSpanId(context.SpanId),
+                ToActivityKind(activity),
+                activity.OperationName,
+                activity.StartTimeUtc.ToEpochMicroseconds(),
+                duration: (long)activity.Duration.TotalMilliseconds * 1000, // duration in Microseconds
+                localEndpoint,
+                remoteEndpoint,
+                annotations,
+                attributeEnumerationState.Tags,
+                null,
+                null);
+        }
+
+        internal static string EncodeSpanId(ActivitySpanId spanId)
+        {
+            return spanId.ToHexString();
+        }
+
+        internal static long ToEpochMicroseconds(this DateTimeOffset timestamp)
+        {
+            return timestamp.ToUnixTimeMilliseconds() * 1000;
+        }
+
+        internal static long ToEpochMicroseconds(this DateTime utcDateTime)
+        {
+            const long TicksPerMicrosecond = TimeSpan.TicksPerMillisecond / 1000;
+            const long UnixEpochTicks = 621355968000000000; // = DateTimeOffset.FromUnixTimeMilliseconds(0).Ticks
+            const long UnixEpochMicroseconds = UnixEpochTicks / TicksPerMicrosecond;
+
+            // Truncate sub-microsecond precision before offsetting by the Unix Epoch to avoid
+            // the last digit being off by one for dates that result in negative Unix times
+            long microseconds = utcDateTime.Ticks / TicksPerMicrosecond;
+            return microseconds - UnixEpochMicroseconds;
+        }
+
+        private static string EncodeTraceId(ActivityTraceId traceId, bool useShortTraceIds)
+        {
+            var id = traceId.ToHexString();
+
+            if (id.Length > 16 && useShortTraceIds)
+            {
+                id = id.Substring(id.Length - 16, 16);
+            }
+
+            return id;
+        }
+
+        private static string ToActivityKind(Activity activity)
+        {
+            switch (activity.Kind)
+            {
+                case ActivityKind.Server:
+                    return "SERVER";
+                case ActivityKind.Producer:
+                    return "PRODUCER";
+                case ActivityKind.Consumer:
+                    return "CONSUMER";
+                default:
```

Comment:
Do we want "CLIENT" for `ActivityKind.Internal`? If yes, might need to comment here.

Reply:
Zipkin does not respect Internal. Java and Python OT exporter return Null for Internal. Used same pattern here.